### PR TITLE
Create main window app & update CMakeLists.txt for Qt

### PR DESCRIPTION
**Pull request by AnWuPP**
1. Create base main window for app
2. Update CMakeLists.txt for Qt

TODO:
1. Update uplaod Qt5 in CMake (maybe use PATH)

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,30 @@
...
+    set(CMAKE_INCLUDE_CURRENT_DIR ON)
+endif()
+
+## Qt components
+set(CMAKE_PREFIX_PATH "C:\\Qt\\5.15.2\\mingw81_64\\lib\\cmake")
+find_package(Qt5 COMPONENTS
+                    Core
+                    Gui
+                    Widgets
+                    WebEngineWidgets
```
Comment (DVTimofeev) [inline CMakeLists.txt:21]: Have you checked if webengine work with mingw compiler?

File: src/gui/mainwindow.cpp [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#include "mainwindow.h"
+MainWindow::MainWindow(QWidget* parent) {
+	ui.setupUi(this);
```
Comment (DVTimofeev) [inline src/gui/mainwindow.cpp:3]: becouse of this automatic UI thing I cannot see and imagine the mainwindow )

File: src/gui/mainwindow.ui [hunk 3]
```diff
@@ -0,0 +1,67 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (DVTimofeev) [inline src/gui/mainwindow.ui:1]: maybe it is reasonable to add mainwindow.hcc (copy C++ code from designer) to at least see what is in it

**Review by DVTimofeev — COMMENTED** [review]
Just overlook on first commit.
see comments

File: CMakeLists.txt (hunk 1 above)
Comment (AnWuPP) [reply CMakeLists.txt:21]: https://wiki.qt.io/Building_Qt_Desktop_for_Windows_with_MinGW
mingw32-make don't work :) Just use qmake for all builds

File: src/gui/mainwindow.cpp (hunk 2 above)
Comment (AnWuPP) [reply src/gui/mainwindow.cpp:3]: Its very bad solution, Your always see ui_mainwindow.h is BUILD_DIR/SEngine_autogen/include dir.

File: src/gui/mainwindow.ui (hunk 3 above)
Comment (AnWuPP) [reply src/gui/mainwindow.ui:1]: And each time change to a manual file... A.. no.

**Pull request merged**